### PR TITLE
Add Siren handlers for sirenAddressed and sirenEscalated (CU-1dffcm4, CU-1ex9q5n)

### PR DESCRIPTION
**Pull request by trisapeace**
- sirenAddressed is called when someone pushes the button on the Siren, this completes the session (webhook is defined here: https://console.particle.io/dev-argons-15515/integrations)
- sirenEscalate is called when the siren times out and turns itself off, this sends a message to the Fallback Phones (webhook is defined here: https://console.particle.io/dev-argons-15515/integrations)
- Factored the Brave Siren code out of `index.js` and `serverTest.js` following the same pattern as when I moved out the Dashboard code
- Moved redis `connect()` and `disconnect()` calls from `serverTest.js` to a global `mochaFixtures.js` file so that it only runs once for all integration tests
- For CU-1ex9q5n, also changed the way that we are calling the `start` function through the Particle API. Now it needs to take the Particle product group so that we can use an API user with reduced permissions and an non-expiring access token. The README explains more about how this was set up and how one should rotate the access token.

## Test Plan
- :heavy_check_mark: Deploy to dev.sensors.brave.coop

- :heavy_check_mark: Run the smoke tests

- :heavy_check_mark: Run all the automated tests using `npm test`, `npm run unittest`, and `npm run integrationtest` to make sure that the mochaFixtures setup and teardown run once and only once
- :heavy_check_mark: Run all the automated tests using the "Current File" and "All Tests" settings in the VSCode debugger to make sure that the mochaFixtures setup and teardown run once and only once

- :heavy_check_mark: Update my real Boron sensor to have `null` as its `sirenParticleId`
   - :heavy_check_mark: Trigger a DURATION and a STILLNESS alert to see that they start the regular chatbot

- :heavy_check_mark: Update my real Boron sensor to have `sirenParticleId` = the real Siren prototype's Particle Core ID
   - :heavy_check_mark: Trigger a DURATION and a STILLNESS alert to see that the Siren starts and a new session starts on the Dashboard with the correct Alert Type
   - :heavy_check_mark: While Siren is on, push the button to see that the Siren stops and the session's state is updated to COMPLETED on the Dashboard
   - :heavy_check_mark: Let the Siren timeout to see that it sends a message to the Fallback Phones
   - :heavy_check_mark:  Trigger an additional alert within the `SUBSEQUENT_ALERT_MESSAGE_THRESHOLD` to see that there is no change to the session
   - :heavy_check_mark: Trigger an additional alert after the `SUBSEQUENT_ALERT_MESSAGE_THRESHOLD` and before the `SESSION_RESET_THRESHOLD`  to see that it updates the session's updatedAt value in the Dashboard
   - :heavy_check_mark: Trigger an additional alert after the `SESSION_RESET_THRESHOLD` to see that a new session is started
   - :heavy_check_mark: Push the button many times in a row to see that `sirenAddressed` is only called once

- :heavy_check_mark: Use Postman to check the validation error logs for `POST /api/sirenAddressed`
- :heavy_check_mark: Use Postman to check the validation error logs for `POST /api/sirenEscalated`

### CODE REVIEW

**Comment by trisapeace** [comment]
Task linked: [CU-1dffcm4 Brave Siren Backend Milestone 2](https://app.clickup.com/t/1dffcm4)

**Comment by trisapeace** [comment]
What is up on `dev.sensors.brave.coop` right now is this same PR but with the Digital Ocean token fixed in Travis. It passed in Travis: https://app.travis-ci.com/github/bravetechnologycoop/BraveSensor-Server/builds/238093880 but this one won't anymore because the token is out-of-date. But the tests still all pass, so it shouldn't stop us from merging it when we're ready.

File: .vscode/launch.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
       "request": "launch",
       "name": "Mocha All within 'test' folder",
       "program": "${workspaceFolder}/node_modules/mocha/bin/_mocha",
-      "args": ["--timeout", "999999", "--colors", "${workspaceFolder}/test/**/*"],
+      "args": ["--timeout", "999999", "--require", "${workspaceFolder}/test/mochaFixtures.js", "--colors", "${workspaceFolder}/test/**/*"],
```
Comment (mariocimet) [inline .vscode/launch.json:13]: is this to link the new mochaFixtures functions with the test files when we run it through the VSCode debugger?

File: README.md [hunk 2]
```diff
@@ -287,3 +287,35 @@ Follow [these instructions](https://www.digitalocean.com/community/tutorials/how
 1. From a shell with kubectl access to the cluster, cd into `BraveSensor-Server` repository and run `kubectl apply -f manifests/redis_storage.yaml`
 2. Install redis from the bitnami/redis chart by running `helm install <environment name>-sensor-redis bitnami/redis --namespace redis --set persistence.storageClass=redis-storage --set usePassword=false --set cluster.enabled=false`
 3. Run `kubectl get svc --namespace=redis` to get the IP of your new redis instance to use as an environment variable
+
+# Particle API access
+
+The Brave Siren functions use the Particle API to trigger the Siren when a `DURATION` or `STILLNESS` event occurs. This requires the access token (`PARTICLE_ACCESS_TOKEN`) for an API user with permission to call functions in the product group (`PARTICLE_PRODUCT_GROUP`).
+
+To generate this token:
```
Comment (mariocimet) [inline README.md:295]: is there a reason to do this other than setting up a new environment?

**Comment by mariocimet** [comment]
looks great, looks like the Branch travis is failing due to authentication for the container registry? not anything that prevents merging

**Review by mariocimet — APPROVED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (trisapeace) [reply .vscode/launch.json:13]: Yes, that's right.

File: README.md (hunk 2 above)
Comment (trisapeace) [reply README.md:295]: Setting up a new environment or if we ever want to rotate the token for better security. In fact, I've written these instructions as though there already is a token in prod that we don't want to break, so it's more for the rotation case.

**Comment by trisapeace** [comment]
Thanks @mariocimet !

**Pull request merged**